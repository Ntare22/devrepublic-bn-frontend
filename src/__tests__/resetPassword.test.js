/* eslint-disable no-console */
/* eslint-disable react/jsx-props-no-spreading */
import React from 'react';
import { mount } from 'enzyme';
import { act } from 'react-dom/test-utils';
import { Provider } from 'react-redux';
import toJson from 'enzyme-to-json';
import reduxStore from '../store';
import ResetPassword from '../components/authentication/resetPassword';

const resetPasswordComponent = (props = {
  location: {
    search: 'http://localhost:3000/password/reset?token=thisismyfaketoken',
  },
  history: {},
  user: {},
  push: jest.fn(),
  resetPasswordAction: jest.fn(),
}) => {
  const component = mount(
    <Provider store={reduxStore}>
      <ResetPassword {...props} />
    </Provider>,
  );
  return component;
};

const originalError = console.error;

beforeAll(() => {
  console.error = jest.fn();
});

afterAll(() => {
  console.error = originalError;
});

describe('resetPassword component tests', () => {
  let component;
  beforeEach(() => {
    component = resetPasswordComponent();
  });
  it('should render resetPassword component', () => {
    expect(component.exists()).toBe(true);
  });

  it('should render resetPassword title', () => {
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should submit a form with all info', async () => {
    const wrapper = component.find('.container');
    expect(wrapper.length).toBe(1);
  });
  it('should submit a form with all info', async () => {
    const wrapper = component.find('.container');
    expect(wrapper.length).toBe(1);
    act(() => {
      const form = wrapper.find('form');
      const event = { preventDefault: jest.fn() };
      form.simulate('submit', event);
      wrapper.find('input').simulate('change', { target: { value: 'Pass123?' } });
    });
  });
});
