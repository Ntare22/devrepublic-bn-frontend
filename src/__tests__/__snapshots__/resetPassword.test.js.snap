// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`resetPassword component tests should render resetPassword title 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <Connect(resetPassword)
    history={Object {}}
    location={
      Object {
        "search": "http://localhost:3000/password/reset?token=thisismyfaketoken",
      }
    }
    push={[MockFunction]}
    resetPasswordAction={[MockFunction]}
    user={Object {}}
  >
    <resetPassword
      history={Object {}}
      location={
        Object {
          "search": "http://localhost:3000/password/reset?token=thisismyfaketoken",
        }
      }
      push={[MockFunction]}
      resetPasswordAction={[Function]}
      user={
        Object {
          "error": null,
          "user": null,
        }
      }
    >
      <div
        className="content"
      >
        <div
          className="container p-5 justify-content-md-center"
        >
          <form
            className="form-check text-center"
            onSubmit={[Function]}
          >
            <div
              className="form-group justify-content-center align-items-center"
            >
              <h1>
                 Reset your password 
              </h1>
              <ErrorAlert
                message=""
                otherClassName="center col-md-4"
                textColor=""
                visible={false}
              >
                <Alert
                  className="text- center col-md-4"
                  closeAriaLabel="Close"
                  color="info"
                  fade={true}
                  isOpen={false}
                  tag="div"
                  transition={
                    Object {
                      "appear": true,
                      "baseClass": "fade",
                      "baseClassActive": "show",
                      "enter": true,
                      "exit": true,
                      "in": true,
                      "mountOnEnter": false,
                      "onEnter": [Function],
                      "onEntered": [Function],
                      "onEntering": [Function],
                      "onExit": [Function],
                      "onExited": [Function],
                      "onExiting": [Function],
                      "tag": "div",
                      "timeout": 150,
                      "unmountOnExit": true,
                    }
                  }
                >
                  <Fade
                    appear={true}
                    baseClass="fade"
                    baseClassActive="show"
                    className="text- center col-md-4 alert alert-info"
                    enter={true}
                    exit={true}
                    in={false}
                    mountOnEnter={false}
                    onEnter={[Function]}
                    onEntered={[Function]}
                    onEntering={[Function]}
                    onExit={[Function]}
                    onExited={[Function]}
                    onExiting={[Function]}
                    role="alert"
                    tag="div"
                    timeout={150}
                    unmountOnExit={true}
                  >
                    <Transition
                      appear={true}
                      enter={true}
                      exit={true}
                      in={false}
                      mountOnEnter={false}
                      onEnter={[Function]}
                      onEntered={[Function]}
                      onEntering={[Function]}
                      onExit={[Function]}
                      onExited={[Function]}
                      onExiting={[Function]}
                      timeout={150}
                      unmountOnExit={true}
                    />
                  </Fade>
                </Alert>
              </ErrorAlert>
              <input
                className="form-control text-center center col-md-4"
                onChange={[Function]}
                pattern="(?=.*\\\\d)(?=.*[A-Z])(?=.*[a-z])(?=.*\\\\W).{8,30}"
                placeholder="New Password"
                required={true}
                type="password"
                value=""
              />
              <div
                className="invalid-feedback text-lg-center"
              >
                Use a password of at least 8 characters: uppercase, lowercase letters, numbers and special characters.
              </div>
            </div>
            <button
              className="btn btn-primary col-md-2"
              type="submit"
            >
              <span
                className="spinner-border-sm"
              />
              Â 
              RESET
            </button>
          </form>
        </div>
      </div>
    </resetPassword>
  </Connect(resetPassword)>
</Provider>
`;
